### ISSUE TITLE
Wazuh FIM - "show files" timeout when monitor lost of files

### ISSUE THREAD

**Issue by lanyi-jozsef** [wazuh/wazuh-kibana-app#1579]
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 3.9.2   | 7.1.1     | 444 |

**Description**
I cannot use show files when lots of files is monitored with FIM.

I increased the timeout in the kibana configs, but i got timeout after 20 sec, or 'Maximum call stack size exceeded'.

My question:
- how i increase the API timeout?

**Steps to reproduce**

1. Go to 'Integrity monitoring'
2. Click on 'show files'
3. see error 'Maximum call stack size exceeded'
4. search with "wp"
5. see the file list
6. click on "Last modified ^ "
7. see error "Wazuh API error: 3021 - Timeout executing API request"

**Screenshots**

![timeout](https://user-images.githubusercontent.com/44770411/60963500-4215b080-a311-11e9-86ff-d4d0d44ef7e2.png)
![maximum](https://user-images.githubusercontent.com/44770411/60963503-43df7400-a311-11e9-899e-eb8d6f26a7fe.png)

**Additional context**
I use docker to run manager

kibana.yml : 
- elasticsearch.requestTimeout: 900000

kibana wazuh plugin:
- timout: 90000

the request time after 20sec

I monitor several WP directory.
I don't monitor the cache files.

**Comment by juankaromo** [comment]
Hi @lanyi-jozsef,

Did you restart Kibana after the timeout parameter change? You need to perform a restart to apply the new settings.

On the other hand, ping @wazuh/framework . Currently the GET /syscheck/{agent_id} call does not work properly due to a failure with the cluster timeout when a lot of files are monitored.

Regards.

**Comment by AdriiiPRodri** [comment]
Hi @lanyi-jozsef and @juankaromo,

To solve this kind of problems we can add the **?wait_for_complete** parameter to the call, this way the timeout is not applied. 

Another solution is to modify the timeout which is found in ```/var/ossec/framework/wazuh/cluster/cluster.json```

Inside we must look for the parameters **timeout_cluster_request** and **timeout_api_exe**.
```json
"communication":{
            "timeout_cluster_request": 20,
            "timeout_api_request": 200,
            "timeout_api_exe": 10,
            "timeout_receiving_file": 120
        }
```
Once the values of **timeout_cluster_request** and **timeout_api_exe** have changed, we must **restart** Wazuh for the changes to be applied correctly.

Regards.

**Comment by juankaromo** [comment]
Thanks @AdriiiPRodri, I take note!

**Comment by juankaromo** [comment]
Hi @lanyi-jozsef,

This has already been fixed in the new version of Wazuh 3.11. This and many other tables have been migrated to React and a Table component has been used that includes a result pagination system that prevents the memory stack from overflowing.

Regards,
